lib/fastlane/plugin/ddg_apple_automation/actions/asana_upload_action.rb

```diff
@@ -0,0 +1,66 @@
+require "fastlane/action"
+require "fastlane_core/configuration/config_item"
+require "httparty"
+require_relative "../helper/ddg_apple_automation_helper"
+
+module Fastlane
+  module Actions
+    class AsanaUploadAction < Action
+      def self.run(params)
+        task_id = params[:task_id]
+        token = params[:asana_access_token]
+        file_name = params[:file_name]
+
+        begin
+          file = File.open(file_name)
+        rescue Errno::ENOENT
+          UI.user_error!("Failed to open file: #{file_name}")
+        end
+
+        url = Helper::DdgAppleAutomationHelper::ASANA_API_URL + "/tasks/#{task_id}/attachments"
+        response = HTTParty.post(url,
+                                 headers: { 'Authorization' => "Bearer #{token}" },
+                                 body: { file: file })
+
+        unless response.success?
+          UI.user_error!("Failed to upload file to Asana task: (#{response.code} #{response.message})")
+        end
```

Comment:
Since `UI.user_error!` doesn't cause a test to stop (while it causes fastlane to stop the action/lane), it's tricky to test `UI.user_error!` stopping execution. I would suggest that we include all this code under begin/rescue clause, so that `UI.user_error!` calls are always the last thing to happen within `#run` function.

This may not scale as the complexity of our action raises and we may need to either find a different solution or stop testing user_error that thoroughly :/